### PR TITLE
fix windows event sink log compilation error with UNICODE preprocessor

### PR DESCRIPTION
**Pull request by iko1**

### CODE REVIEW

**Comment by gabime** [comment]
You got me confused. I see you removed the ```#ifdef SPDLOG_WCHAR_TO_UTF8_SUPPORT```
 So what's the purpose of this pr ?

**Comment by iko1** [comment]
I updated this PR. I'm using #ifdef to check if `SPDLOG_WCHAR_TO_UTF8_SUPPORT` and then convert the message accordingly before calling to UNICODE version of `ReportEvent`.

File: include/spdlog/sinks/win_eventlog_sink.h [hunk 1]
```diff
@@ -218,13 +220,23 @@ class win_eventlog_sink : public base_sink<Mutex>
...
+        bool succeeded;
         memory_buf_t formatted;
         base_sink<Mutex>::formatter_->format(msg, formatted);
         formatted.push_back('\0');
-        LPCSTR lp_str = static_cast<LPCSTR>(formatted.data());
 
-        auto succeeded = ::ReportEvent(event_log_handle(), eventlog::get_event_type(msg), eventlog::get_event_category(msg), event_id_,
-            current_user_sid_.as_sid(), 1, 0, &lp_str, nullptr);
+#ifdef SPDLOG_WCHAR_TO_UTF8_SUPPORT
+        auto buf = std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(formatted.data());
```
Comment (gabime) [inline include/spdlog/sinks/win_eventlog_sink.h:229]: ```std::wstring_convert<std::codecvt_utf8>``` is deprecated. Please use  ```details::os::wstr_to_utf8buf(..)``` instead. See usage example [here](https://github.com/gabime/spdlog/blob/9ce39a470fa7adc8b101fdd5e4325e6c371f675c/include/spdlog/logger.h#L243).

File: include/spdlog/details/os-inl.h [hunk 2]
```diff
@@ -459,6 +459,41 @@ SPDLOG_INLINE void wstr_to_utf8buf(wstring_view_t wstr, memory_buf_t &target)
...
+
+        if (result_size > 0)
+        {
+            target.resize(result_size);
+            return;
+        }
+    }
+
+    throw_spdlog_ex(fmt::format("WideCharToMultiByte failed. Last error: {}", ::GetLastError()));
+}
```
Comment (gabime) [inline include/spdlog/details/os-inl.h:496]: wrong error message in exception. should be “ MultiByteToWideChar”

**Review by gabime — CHANGES_REQUESTED** [review]

**Pull request merged**